UI to support selecting user role/type when signing up

shalandy opened the pull request:
### (new commit)

### Demo
![c19bayshield_userdb](https://user-images.githubusercontent.com/25168261/79397129-3acfa300-7f32-11ea-89ac-ee824c6be47f.gif)

### Edited .dart files:
`login_screen`: create new document in users collection upon valid signup
`maker_screen`: basic stateful screen with appbar and TODO content
`root_screen`: using switch statements based on user type to navigate to correct screen: hub, maker, or consumer
`user`: allow converting to and from firestore and local data model formats; add id and type field

### New .dart files:
`hub_screen`: basic stateful screen with appbar and TODO content
`locator`: use Getit to use _firestoreUser as a service/util
`firestore_users`: api that links user data model with users collection from firestore; allows for creating users, updating user type, and getting user from id

> Note: added config info for iOS firebase app but need cocaopod installation via xcode

### Demo
![c19bayshield_userselect](https://user-images.githubusercontent.com/25168261/79172440-902c7880-7da9-11ea-8a14-747b167b2cc5.gif)

> Note: no changes on User data model yet (did not create type/role field) 

### Edited screens:
`login_screen`: automatically login at signup and redirect after to choose user type/role

### New screens:
`user_type_screen`: page with user type/role selections and finishes login callback after user selects one, redirecting them to `consumer_screen` (for now)

shalandy committed:
UI to support selecting user role/type when signing up

cameronaaron reviewed (approved):
LGTM but lets mark this as don't merge yet until we have backend working

shalandy committed:
Link local user model to firestore with navigation based on userType

### Edited .dart files:
`login_screen`: create new document in users collection upon valid signup
`maker_screen`: basic stateful screen with appbar and TODO content
`root_screen`: using switch statements based on user type to navigate to correct screen: hub, maker, or consumer
`user`: allow converting to and from firestore and local data model formats; add id and type field

### New .dart files:
`hub_screen`: basic stateful screen with appbar and TODO content
`locator`: use Getit to use _firestoreUser as a service/util
`firestore_users`: api that links user data model with users collection from firestore; allows for creating users, updating user type, and getting user from id

> Note: added config info for iOS firebase app but need cocaopod installation via xcode

cameronaaron committed:
IOS init

cameronaaron commented:
Is this ready to go?

tinataleb commented:
looks great! amazing job @shalandy 
the only thing I would prefer to be changed is the order.
It makes more sense for someone to pick what type they are signing up as before being asked to enter email and password.
Is it possible to change the order to what I mentioned above?
